wicket-core/src/main/java/org/apache/wicket/resource/FileSystemResource.java

```diff
@@ -95,7 +95,9 @@ protected ResourceResponse createResourceResponse(Path path, String fileName)
 			resourceResponse.setContentType(getMimeType());
 			resourceResponse.setAcceptRange(ContentRangeType.BYTES);
 			resourceResponse.setContentLength(size);
-			resourceResponse.setFileName(fileName == null ? path.getFileName().toString() : fileName);
+			if (path != null && path.getFileName() != null) {
```

Comment:
I think we don't need to check *path != null* because it is checked before (and a WicketRuntimeException is thrown if it is null) - everything else looks good.